### ISSUE TITLE
It is difficult to debug misconfigured .nixd.json

### ISSUE THREAD

**Issue by lf-** [nix-community/nixd#266]
The point of this issue is not so much that my configuration doesn't work, but that it is very hard to figure out why, as I can't seem to find any useful logging of why the evaluation (may have) failed. I have had this issue in the past, which is why I am filing a bug this time to record it as something that's hard.

I am currently working on a project with a `.nixd.json` of the following, and am not getting options completion, but I don't know why:

```json
{
    "options": {
        "enable": true,
        "target": {
            "installable": "",
            "args": ["--impure", "-f", "vm.nix", "nixosConfigurations.boxMicro.options"]
        }
    }
}
```

vm.nix contains the following:

```nix
let flake = builtins.getFlake "/home/jade/co/microvm.nix";
in
{
  nixosConfigurations.boxMicro = flake.inputs.nixpkgs.lib.nixosSystem {
    system = "x86_64-linux";
    modules = [
      flake.nixosModules.microvm
      ./vmcfg.nix
    ];
  };
}
```

 I have set the following options on the server in coc.nvim (I found the log level option by reading the source code):

```json
{
    "languageserver": {
        "nix": {
            "command": "nixd",
            "args": ["-log=debug"],
            "filetypes": ["nix"],
            "settings": {},
        }
}
```

However, the system evaluates, so something is going wrong along the way:

```
co/microvm.nix » nix eval --impure -f vm.nix nixosConfigurations.boxMicro.config.system.build.to
plevel.outPath  
trace: warning: system.stateVersion is not set, defaulting to 23.11. Read why this matters on ht
tps://nixos.org/manual/nixos/stable/options.html#opt-system.stateVersion.
"/nix/store/fgzjygyacafqz9v79hfxizmrk8azdpib-nixos-system-nixos-23.11.20230831.3e52e76"
```

There is a pile of logs here: https://gist.github.com/lf-/1b3b3cc5627b91f8945d8bbe7ba6dca2

Notable parts:

```
E[15:28:05.196] 3364889: Failed to decode workspace/configuration reply: expected object
V[15:28:05.196] 3364889: /* error: expected object */
[
  null
]
V[15:28:05.196] 3364889: <<< {"jsonrpc":"2.0","id":2,"result":[null]}
I[15:28:05.196] 3364889: <-- reply(2)
E[15:28:05.196] 3364889: Failed to decode workspace/configuration reply: expected object
V[15:28:05.196] 3364889: /* error: expected object */
[
  null
]

V[15:28:12.343] 3364889: worker 0 reported error: -32001: no AST available on requested file /home/jade/co/microvm.nix/nixos-modules/microvm/optimization.nix
```

The issue here, though, is that there is voluminous amounts of logging but nothing of why nix is failing to evaluate the options or anything like that, which is what I would need to figure out why my workspace config is wrong.

**Comment by evelant** [comment]
I'm having a similar problem. Trying to get nixd working with a flake based `nix-darwin` and `home-manager` setup to complete options. I haven't been able to figure it out.

**Comment by inclyc** [comment]
> The issue here, though, is that there is voluminous amounts of logging but nothing of why nix is failing to evaluate the options or anything like that, which is what I would need to figure out why my workspace config is wrong.

I've just read the log, and looks like your workspace configuration should work as expected? 

Note that our module completion system does not properly support nested attrs due to frontend issue(https://github.com/nix-community/nixd/pull/249), and it also cannot deal with submodules (#106).

#249 should be completely addressed by #257 + #262 , #106 is dependent on these two important PRs.
